### DISCUSSION TITLE
I am getting an error when library uses the Logical Or Statement (||=)

### DISCUSSION THREAD

**Discussion by nurullah** [vercel/next.js#41445] [Help] [answered]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```
Operating System:
      Platform: linux
      Arch: x64
      Version: #45~20.04.1-Ubuntu SMP Mon Apr 4 09:38:31 UTC 2022
    Binaries:
      Node: 14.19.3
      npm: 6.14.17
      Yarn: 1.22.19
      pnpm: N/A
    Relevant packages:
      next: 12.3.2-canary.23
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
```

### What browser are you using? (if relevant)

Google Chrome Version 104.0.5112.79 (Official Build) (64-bit)

### How are you deploying your application? (if relevant)

next dev

### Describe the Bug

I was created the next.js application via command line. And i installed the pdfjs-dist library. I would continue to work but i get an error about the Logical OR Statement.

```js
import { PDFWorker } from "pdfjs-dist";
const worker = new PDFWorker("custom-worker");
```
If i replace the above code like the below, i  don't get any error about Logical OR Assignment.
```js
let status = false;
status ||= 'close';
```

### Expected Behavior

I was hoping not to get any errors.

### Link to reproduction

https://codesandbox.io/s/nextjs-logical-or-assignment-psfd02

### To Reproduce

1. Click the codesandbox link
2. See the "`SyntaxError: Unexpected token '||='`" error
3. Replace the 3-4 line of the `pages/index.js` like the below
```js
let status = false;
status ||= 'close';
```
4. See the problem is only about the importing a library that uses the logical error assignment.

**Comment by SukkaW** [comment, marked as answer]
Next.js' compiler currently ignores all the files from node_modules. If the library uses the unsupported syntax, it will not be transpiled away by Next.js' compiler.

You can use a community maintained plugin `next-transpile-modules` for now: https://www.npmjs.com/package/next-transpile-modules
